lessons/taquito/06/06.mdx

```diff
@@ -0,0 +1,136 @@
+---
+title: Originating, finally.
+chapter: Chapter 6
+slug: module-04/chapter-06
+filterBy: lesson-4
+editor:
+  language: javascript
+  startingCode: |
+    const FAUCET_KEY = {
+        "mnemonic": [
+            "chimney",
+            "item",
+            "course",
+            "pair",
+            "opinion",
+            "mass",
+            "receive",
+            "wrong",
+            "hurry",
+            "nest",
+            "oil",
+            "acoustic",
+            "section",
+            "salad",
+            "electric"
+        ],
+        "secret": "47184b1d8bf4dd550256817c3507cc4ccf112888",
+        "amount": "903205016",
+        "pkh": "tz1Q654L2qnt5mDM1Lbi4rMSTtnE5JHvD8M7",
+        "password": "9U5VOADM6l",
+        "email": "ryqwtuyj.ciwjnvdt@tezos.example.org"
+    }
+    
+    importKey(
+        Tezos,
+        FAUCET_KEY.email,
+        FAUCET_KEY.password,
+        FAUCET_KEY.mnemonic.join(' '),
+        FAUCET_KEY.secret
+    );
+    
+    Tezos.setProvider({ rpc: 'https://carthagenet.SmartPy.io' });
+    
+    // const CODE_JSON = 
```

Comment:
I think using the below code as a template and asking users to fill in the details would make it much smoother. Considering our editor isn't exactly an editor.

```javascript
Tezos.contract.originate({
        code: // add CODE_JSON here,
        init: // add STORAGE_JSON here
    }).then(origination => {
        print(`Origination waiting confirmation...`);
        return origination.contract()
    }).then(contract => {
        print(`Origination completed.`);
        print(`Contract: ${contract.address}`)
    })
```